x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/authn/UserServiceAuthentication.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.idp.saml.authn;
+
+import org.elasticsearch.xpack.idp.saml.sp.SamlServiceProvider;
+
+import java.util.Set;
+
+/**
+ * Lightweight representation of a user that has authenticated to the IdP in the context of a specific service provider
+ */
+public interface UserServiceAuthentication {
+
+    String getPrincipal();
+    Set<String> getGroups();
```

Comment:
Yeah, I was torn on this. They're not really groups, but `roles` is so over used that I think it will become confusing.

From an ES point of view they'll probably be something related to application privileges.
So, from that position they're `privileges`, but I'm not sure whether exposing that name here is right...

Reply:
We can defer the decision to a later point I guess